src/config.rs

```diff
@@ -0,0 +1,248 @@
+    pub s3_force_path_style_destination: Option<bool>,
+
+    // /// The access key ID for the S3 bucket.
+    // #[arg(long, env = "S3_ACCESS_KEY_ID_SOURCE", requires_all = ["s3_secret_access_key_source"])]
+    // pub s3_access_key_id_source: Option<String>,
+
+    // /// The secret access key for the S3 bucket.
+    // #[arg(long, env = "S3_SECRET_ACCESS_KEY_SOURCE", requires_all = ["s3_access_key_id_source"])]
+    // pub s3_secret_access_key_source: Option<String>,
+
+    // /// The session token for the S3 bucket.
+    // #[arg(long, env = "S3_SESSION_TOKEN_SOURCE", requires_all = ["s3_access_key_id_source", "s3_secret_access_key_source"])]
+    // pub s3_session_token_source: Option<String>,
+    /// The access key ID for the S3 bucket.
+    #[arg(long, env = "S3_ACCESS_KEY_ID_DESTINATION", requires_all = ["s3_secret_access_key_destination"])]
+    pub s3_access_key_id_destination: Option<String>,
+
+    /// The secret access key for the S3 bucket.
+    #[arg(long, env = "S3_SECRET_ACCESS_KEY_DESTINATION", requires_all = ["s3_access_key_id_destination"])]
+    pub s3_secret_access_key_destination: Option<String>,
+
+    /// The session token for the S3 bucket.
+    #[arg(long, env = "S3_SESSION_TOKEN_DESTINATION", requires_all = ["s3_access_key_id_destination", "s3_secret_access_key_destination"])]
+    pub s3_session_token_destination: Option<String>,
+
+    // todo: add --force option
+    #[command(flatten)]
+    pub verbose: Verbosity,
+}
+
+#[derive(Clone)]
+pub struct S3Credentials {
+    pub access_key_id: String,
+    pub secret_access_key: String,
+    pub session_token: Option<String>,
+}
+
+/* -------------------------------------------- YAML ------------------------------------------- */
+
+#[derive(Debug, Clone)]
+pub struct GlobPattern(glob::Pattern);
+
+impl<'de> Deserialize<'de> for GlobPattern {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        glob::Pattern::from_str(&s)
+            .map(GlobPattern)
+            .map_err(serde::de::Error::custom)
+    }
+}
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(untagged)]
+pub enum PackageConfig {
+    PackageGlob {
+        name: GlobPattern,
+        version: Option<VersionSpec>,
+    },
+    License {
+        license: String,
+    },
+}
+
+impl PackageConfig {
+    pub(crate) fn matches(&self, package_record: PackageRecord) -> bool {
+        match self {
+            PackageConfig::PackageGlob { name, version } => {
+                let name_match = name.0.matches(package_record.name.as_normalized());
+                if let Some(version) = version {
+                    name_match && version.matches(package_record.version.as_ref())
+                } else {
+                    name_match
+                }
+            }
+            PackageConfig::License { license } => package_record
```

Comment:
I wonder if this is necessary. Technically a MatchSpec could do all that and more for you (e.g. `*[license="foobar"]`).

Reply:
does a matchspec also support globs in package names?

i.e. `ros-*=*=*`?

Reply:
It does not support globs in package names afaik. But you can use the NamelessMatchSpec, maybe?

Reply:
depends on https://github.com/conda/rattler/pull/1236

Reply:
it works right now but `[license="MIT"]` doesn't work until https://github.com/conda/rattler/pull/1236 is through. i wouldn't wait for this though but just create a second release afterwards with a newer rattler version once available